definitions/tools/merge_somatic_vcfs.cwl

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env cwl-runner
+
+cwlVersion: v1.0
+class: CommandLineTool
+label: "merge several somatic VCFs into a multi-sample VCF"
+baseCommand: ["/bin/bash", "merge_somatic_vcfs.sh"]
+
+requirements:
+    - class: ResourceRequirement
+      ramMin: 4000
+    - class: DockerRequirement
+      dockerPull: 'zlskidmore/samtools:1.10'
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: 'merge_somatic_vcfs.sh'
+        entry: |
+            set -eou pipefail
+
+            vcfs="$1" #csv list of vcfs
+            samples="$2" #csv list of tumor sample names
+            normal_sample_name="$3"
+            
+            IFS=',' read -ra vcf_array <<< "$vcfs"
+            IFS=',' read -ra sample_array <<< "$samples"
+            
+            if [ "${#vcf_array[@]}" != "${#sample_array[@]}" ]; then
+                echo "ERROR: the number of vcfs must be equal to the number of sample names provided";
+                exit 1
+            fi
+            
+            # prep all the tumor vcfs for merging
+            for i in ${!vcf_array[*]}; do
+                samp=${sample_array[$i]};
+                vcf=${vcf_array[$i]};
+                #assumes the input VCF contains the samples in the order NORMAL TUMOR
+                normal_name_in_vcf=`gunzip -c "$vcf" | grep "^#CHROM" | cut -f 10`;
+                tumor_name_in_vcf=`gunzip -c "$vcf" | grep "^#CHROM" | cut -f 11`;
+                #use index for name of temp files (instead of sample names) to avoid potential weird character issues
+                /usr/local/bin/bcftools view -s "$tumor_name_in_vcf" "$vcf" | /usr/local/bin/bcftools reheader -s <(echo "$samp") | /usr/local/bin/bgzip -c >"$i.tumor_only.vcf.gz";
+                /usr/local/bin/tabix -p vcf "$i.tumor_only.vcf.gz";
+                
+                #also create normal-only vcfs
+                /usr/local/bin/bcftools view -s "$normal_name_in_vcf" "$vcf" | /usr/local/bin/bcftools reheader -s <(echo "$normal_sample_name") | /usr/local/bin/bgzip -c >"$i.normal_only.vcf.gz";
+                /usr/local/bin/tabix -p vcf "$i.normal_only.vcf.gz";
+            done
+                
+            # #create a merged normal vcf that includes all variants from all samples - this isn't strictly necessary,
+            # #but gives us a more sane final vcf
+            # /usr/local/bin/bcftools concat -a -D *.normal_only.vcf.gz | /usr/local/bin/bgzip -c >merged.normal.prenorm.vcf.gz
+            # /usr/local/bin/tabix -p vcf merged.normal.prenorm.vcf.gz
+            # #also have to normalize to prevent an edge case where multiallelic sites are split across multiple lines
+            # /usr/local/bin/bcftools annotate -x FORMAT/F1R2,FORMAT/F2R1 merged.normal.prenorm.vcf.gz | /usr/local/bin/bcftools norm -m+ | /usr/local/bin/bgzip -c >merged.normal.vcf.gz
+            # /usr/local/bin/tabix -p vcf merged.normal.vcf.gz
```

Comment:
What are these newly-added commented-out lines for?

Reply:
Sorry - I didn't mean to push this yet. still testing/tweaking.  If I could find the right button, I'd mark this as a draft...

Reply:
On desktop it's under the reviewers in the righthand column.